### ISSUE TITLE
Unable to change format on non TTY output streams

### ISSUE THREAD

**Issue by drock** [jfelchner/ruby-progressbar#102]
If you are running code using the progress bar using a non TTY output (running a rake task from within RubyMine for example), the format is always the default format.

It appears as though Outputs::NonTty has never been fully implemented (at least for formatting).

```ruby
bar = ProgressBar.create(format: '%j')
bar.format = '%j'
bar.progress = 50
```

Neither of the first two lines will change the format when $stdout.tty? returns false

**Comment by jfelchner** [comment]
@drock format can't be changed on non-TTY output. Check the wiki for why. There's a non-TTY section.

**Comment by jfelchner** [comment]
@drock [a new feature](https://github.com/jfelchner/ruby-progressbar/wiki/Custom-Output-Classes) being released in 1.9 might be of some interest to you.  It should allow you to subclass `ProgressBar::Outputs::NonTty` and override the format to be whatever you would like.  I don't think it's going to behave as you'd like (non-TTY is weird), but I thought I'd throw it in here in case you want to give it a shot.
